### PR TITLE
MOTOR-378 Bump minimum PyMongo version to 3.9

### PR DESCRIPTION
**Pull request by prashantmital**
Closes https://jira.mongodb.org/browse/MOTOR-378

### CODE REVIEW

File: doc/requirements.rst [hunk 1]
```diff
@@ -54,25 +54,29 @@ Motor and PyMongo
...
++---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+| Motor Version | 1.0 |  Y  |  Y  |  Y  |  Y  |  Y  |**N**|**N**|**N**|**N**|
++---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+|               | 1.1 |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |**N**|**N**|**N**|
++---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+|               | 1.2 |**N**|**N**|  Y  |  Y  |  Y  |  Y  |  Y  |**N**|**N**|
++---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+|               | 1.3 |**N**|**N**|  Y  |  Y  |  Y  |  Y  |  Y  |**N**|**N**|
++---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+|               | 2.0 |**N**|**N**|**N**|  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |
```
Comment (prashantmital) [inline doc/requirements.rst]: @ajdavis any idea why we say 2.6 is not supported by Motor, when it is definitely supported by the PyMongo versions these Motor releases use?

**Comment by prashantmital** [comment]
Seems pointless to run a patch for this since all of EVG is already testing with most-recent PyMongo that satisfies `['pymongo>=3.6,<4']` which is 3.9.0

File: doc/requirements.rst (hunk 1 above)
Comment (ajdavis) [reply doc/requirements.rst]: Interesting. I decided this deliberately but I can't remember why:

https://jira.mongodb.org/browse/MOTOR-252

File: doc/requirements.rst [hunk 2]
```diff
@@ -103,6 +107,8 @@ known to be incompatible, or have not been tested together.
 +---------------+-----+-----+-----+-----+
 |               | 2.0 |**N**|  Y  |  Y  |
 +---------------+-----+-----+-----+-----+
+|               | 2.1 |**N**|  Y  |  Y  |
++---------------+-----+-----+-----+-----+
```
Comment (ajdavis) [inline doc/requirements.rst]: Let's add a Tornado 6 column, and start testing with Tornado 6 if we haven't already?

File: doc/requirements.rst [hunk 3]
```diff
@@ -54,25 +54,29 @@ Motor and PyMongo
...
-|               | 1.1 |  Y  |  Y  |  Y  |  Y  |  Y  |  Y  |**N**|**N**|
-+---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+
-|               | 1.2 |**N**|**N**|  Y  |  Y  |  Y  |  Y  |  Y  |**N**|
-+---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+
-|               | 1.3 |**N**|**N**|  Y  |  Y  |  Y  |  Y  |  Y  |**N**|
-+---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+
-|               | 2.0 |**N**|**N**|**N**|  Y  |  Y  |  Y  |  Y  |  Y  |
-+---------------+-----+-----+-----+-----+-----+-----+-----+-----+-----+
++---------------------------------------------------------+-----+-----+-----+
+|                    MongoDB Version                      |     |     |     |
```
Comment (ajdavis) [inline doc/requirements.rst]: Let's remove the extra |'s on the right, this line needs a | character only at the beginning and end.

File: doc/requirements.rst (hunk 1 above)
Comment (prashantmital) [reply doc/requirements.rst]: Let's leave it be for now. I doubt anyone is using ancient Mongo with a cutting-edge driver.

File: doc/requirements.rst (hunk 2 above)
Comment (prashantmital) [reply doc/requirements.rst]: Done.

File: doc/requirements.rst (hunk 3 above)
Comment (prashantmital) [reply doc/requirements.rst]: Done.

File: doc/requirements.rst [hunk 4]
```diff
@@ -130,6 +136,8 @@ Motor versions.
 +---------------+-----+-----+-----+-----+-----+-----+-------+-------+-----+-----+
 |               | 2.0 |**N**|**N**|  Y  |**N**|  Y  |**N**  |  Y    |  Y  |  Y  |
 +---------------+-----+-----+-----+-----+-----+-----+-------+-------+-----+-----+
+|               | 2.1 |**N**|**N**|  Y  |**N**|  Y  |**N**  |  Y    |  Y  |  Y  |
```
Comment (prashantmital) [inline doc/requirements.rst]: @ajdavis why do we break out 3.5 into 3.5.0 and 3.5.2? We don't specify micro version numbers for any other Python runtime version... Should I consolidate this into a single 3.5 column?

**Comment by prashantmital** [comment]
@ajdavis This is ready for another look.

**Comment by prashantmital** [comment]
Tests work as intended, though they do not pass. https://evergreen.mongodb.com/version/5d718dcad6d80a55c2a5dcf6

File: doc/requirements.rst (hunk 4 above)
Comment (behackett) [reply doc/requirements.rst]: https://github.com/mongodb/motor/commit/d49ff6551a95c8b5dc0bcae1096dda8010e80492#diff-f53de7fd54b727d055c43104e20d05f7

Comment (behackett) [reply doc/requirements.rst]: See the notes above the table.

Comment (prashantmital) [reply doc/requirements.rst]: Ah, my bad. Thanks for pointing it out.

**Comment by prashantmital** [comment]
Merged in https://github.com/mongodb/motor/commit/64adf522366235c55da31d621d91e269775bf5ca

**Pull request closed without merging**